### ISSUE TITLE
What's the right way to convert Java collections to native Python collections?

### ISSUE THREAD

**Issue by naddeoa** [py4j/py4j#518]
When sending collections from Java to Python, the thing that you end up getting in Python is a `py4j.java_collections.JavaList` (for lists). These do behavior mostly like Python lists but there are cases where they don't, like serialization/pickling.

Is there an easy/built in way of converting the py4j collection types into native Python collections or do we just have to manually convert them?

**Comment by HyukjinKwon** [comment]
Maybe just invoke `list(py4j.java_collections.JavaList)` ?

**Comment by naddeoa** [comment]
@HyukjinKwon yes, that is how you would start to manually convert them, but you'll have to make a bunch of utility functions that handle recursive data structures as well. That won't work for a list of lists, for example. That's fine if there is no built in utility, I'll just have to write all of that code, but I'm hoping that isn't the case.
